packages/devtools_app/lib/src/flutter/app.dart

```diff
@@ -297,32 +272,13 @@ class DevToolsAboutDialog extends StatelessWidget {
 
 // TODO(devoncarew): Add an analytics setting.
 
-// TODO(devoncarew): Convert the SettingsDialog over to using a controller?
-// Add a settings controller to Controllers that Widgets can access via
-// Controllers.of(context); the SettingsController could manage setting states
-// (where setting values are ValueNotifiers) and have methods to modify them.
-// Widgets (like below) can then just grab the active SettingsController from
-// Controller.of(context).settings and listening to its notifiers.
-
-class SettingsDialog extends StatefulWidget {
-  const SettingsDialog({
-    Key key,
-  }) : super(key: key);
-
-  @override
-  _SettingsDialogState createState() => _SettingsDialogState();
-}
-
-class _SettingsDialogState extends State<SettingsDialog> {
+class SettingsDialog extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
-    void _toggleTheme([bool value]) {
-      value ??= !devtools_theme.isDarkTheme;
-      setState(() {
-        devtools_theme.useDarkTheme = value;
+    final preferences = DevToolsApp.of(context).preferences;
 
-        DevToolsApp.of(context).updateTheme();
-      });
+    void _toggleTheme() {
+      preferences.darkModeTheme.value = !preferences.darkModeTheme.value;
```

Comment:
instead of having the view change this, let's have the view call the controller, which changes the value of darkModeTheme.  We should also only expose the value listenable from the controller and have the notifier be private